File: src/main/java/org/app/app1/scala/PointRunner.scala
package org.app.app1.scala

import org.app.app1.java.Point

// This uses Java Class in Scala Runner

object PointRunner {
  def main(args: Array[String]): Unit = {
    println("Scala Runner")
    val point = new Point(5, 10)
    point.move(5, 10)
    println(point)
  }
}

File: src/main/java/org/app/app2/java/PointRunner.java
package org.app.app2.java;

import org.app.app2.scala.Point;

public class PointRunner {
    public static void main(String ... args) {
        System.out.println("Java Runner");
        Point point = new Point(10, 20);
        point.move(5, 10);
        System.out.println(point);
    }
}